content/tutorial-air-quality-analysis.md

```diff
@@ -0,0 +1,360 @@
+
+The `compute_indices` function first fetches the correct upper and lower bounds of AQI categories and breakpoint concentrations for the input concentration and pollutant with the help of arrays `AQI` and `breakpoints` we created above. Then, it feeds these values into the formula to calculate the sub-index.
+
+```{code-cell} ipython3
+def compute_indices(pol, con):
+    bp = breakpoints[pol]
+    
+    if pol == 'CO':
+        inc = 0.1
+    else:
+        inc = 1
+    
+    if bp[0] <= con < bp[1]:
+        Bl = bp[0]
+        Bh = bp[1] - inc
+        Ih = AQI[1] - inc
+        Il = AQI[0]
+
+    elif bp[1] <= con < bp[2]:
+        Bl = bp[1]
+        Bh = bp[2] - inc
+        Ih = AQI[2] - inc
+        Il = AQI[1]
+
+    elif bp[2] <= con < bp[3]:
+        Bl = bp[2]
+        Bh = bp[3] - inc
+        Ih = AQI[3] - inc
+        Il = AQI[2]
+
+    elif bp[3] <= con < bp[4]:
+        Bl = bp[3]
+        Bh = bp[4] - inc
+        Ih = AQI[4] - inc
+        Il = AQI[3]
+
+    elif bp[4] <= con < bp[5]:
+        Bl = bp[4]
+        Bh = bp[5] - inc
+        Ih = AQI[5] - inc
+        Il = AQI[4]
+
+    elif bp[5] <= con:
+        Bl = bp[5]
+        Bh = bp[5] + bp[4] - (2 * inc)
+        Ih = AQI[6]
+        Il = AQI[5]
+
+    else:
+        print("Concentration out of range!")
+        
+    return ((Ih - Il) / (Bh - Bl)) * (con - Bl) + Il
+```
+
+We will use [np.vectorize](https://numpy.org/devdocs/reference/generated/numpy.vectorize.html) to utilize the concept of vectorization. This simply means we don't have loop over each element of the pollutant array ourselves. [Vectorization](https://numpy.org/devdocs/user/whatisnumpy.html#why-is-numpy-fast) is one of the key advantages of NumPy.
+
+```{code-cell} ipython3
+vcompute_indices = np.vectorize(compute_indices)
+```
+
+By calling our vectorized function `vcompute_indices` for each pollutant, we get the sub-indices. To get back an array with the original shape, we use [np.stack](https://numpy.org/devdocs/reference/generated/numpy.stack.html).
+
+```{code-cell} ipython3
+sub_indices = np.stack((vcompute_indices('PM2.5', pollutants[..., 0]),
+                        vcompute_indices('PM10', pollutants[..., 1]),
+                        vcompute_indices('NO2', pollutants[..., 2]),
+                        vcompute_indices('NH3', pollutants[..., 3]),
+                        vcompute_indices('SO2', pollutants[..., 4]),
+                        vcompute_indices('CO', pollutants[..., 5]),
+                        vcompute_indices('O3', pollutants[..., 6])), axis=1)
+```
+
+### Air quality indices
+
+Using [np.amax](https://numpy.org/devdocs/reference/generated/numpy.amax.html), we find out the maximum sub-index for each period, which is our Air Quality Index!
+
+```{code-cell} ipython3
+aqi_array = np.amax(sub_indices, axis=1)
```

Comment:
I'd vote to just use `max`, which should give the same result, but has a more straightforward name.

```suggestion
aqi_array = np.max(sub_indices, axis=1)
```